src/test/java/com/multithreading/TextProcessorTest.java

```diff
@@ -0,0 +1,48 @@
+package com.multithreading;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+public class TextProcessorTest {
```

Comment:
FYI it's *very* hard to test multi-threaded code. In this case you test the outputs. But if you wanted to test behavior like deadlocks, synchronized behavior or interrupts, it would be super complicated. I recommend you read up on it ;)

Reply:
leaving the tests as-is for now, i'll read up on it later :)